src/htm/encoders/SimHashDocumentEncoder.hpp

```diff
@@ -41,6 +41,23 @@ namespace htm {
      */
     UInt activeBits = 0u;
 
+    /**
+     * @param :charFrequencyCeiling: If param `tokenSimilarity` is on, this will
+     *  be the max number of times a char/letter can be repeated in a token.
+     *  Occurances of the character beyond this number will be discarded.
+     *  A setting of 1 will act as character de-duplication, guaranteeing each
+     *  character in a token is unique. Inverse to param `charFrequencyFloor`.
+     */
+    UInt charFrequencyCeiling = 0u;
+
+    /**
+     * @param :charFrequencyFloor: If param `tokenSimilarity` is on, and if this
+     *  option is set, a character/letter will be ignored until it occurs this
+     *  many times in the token. Occurances of the character before this number
+     *  will be discarded. Inverse to param `charFrequencyCeiling`.
+     */
+    UInt charFrequencyFloor = 0u;
```

Comment:
I like the tokenFreqCeiling/Floor, I'm not sure of usefulness of charFreqFloor ? Note, charFreqCeil is useful.

Reply:
great call, combined the params, and got rid of charFloor: https://github.com/htm-community/htm.core/commit/1d1761159c155851965c0258dba7e8744cf49507